app/assets/javascripts/application.js

```diff
@@ -11,6 +11,12 @@
 // about supported directives.
 //
 //= require rails-ujs
+//= require materialize-sprockets
 //= require activestorage
 //= require turbolinks
 //= require_tree .
+
+document.addEventListener('DOMContentLoaded', function() {
+  var elems = document.querySelectorAll('.sidenav');
+  var instances = M.Sidenav.init(elems, options);
+});
```

Comment:
The `application.js` is only for importing other JS files, can't have any logic here, please move this to a new file and require it here.